File: Programmers/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort()
    l, r = 0, len(people) - 1

    while l < r:
        if people[l] + people[r] <= limit:
            answer += 1
            l += 1
        r -= 1
    return len(people) - answer

File: Programmers/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    int l = 0;
    int r = people.size()-1;
    sort(people.begin(), people.end());

    while (l < r) {
        if (people[l] + people[r--] <= limit) {
            answer += 1;
            l++;
        }
    }
    return people.size() - answer;
}